src/core/jit/input_resolver.rs

```diff
@@ -0,0 +1,68 @@
+use async_graphql_value::{ConstValue, Value};
+
+use super::{ExecutionPlan, ResolveInputError, Variables};
+
+/// Trait to represent conversion from some dynamic type (with variables)
+/// to the resolved variant based on the additional provided info.
+/// E.g. conversion from [async_graphql_value::Value] ->
+/// [async_graphql_value::ConstValue]
+pub trait InputResolvable {
+    type Output;
+
+    fn resolve(
+        self,
+        variables: &Variables<Self::Output>,
+    ) -> Result<Self::Output, ResolveInputError>;
+}
+
+impl InputResolvable for Value {
+    type Output = ConstValue;
+
+    // TODO:
+    // - provide default values
+    fn resolve(self, variables: &Variables<ConstValue>) -> Result<Self::Output, ResolveInputError> {
+        self.into_const_with(|name| {
+            variables
+                .get(&name)
+                .cloned()
+                .ok_or_else(|| ResolveInputError::VariableIsNotFound(name.to_string()))
+        })
+    }
+}
+
+/// Transforms [ExecutionPlan] values the way that all the input values
+/// are transformed to const variant with the help of [InputResolvable] trait
+pub struct InputResolver<Input> {
+    plan: ExecutionPlan<Input>,
+}
+
+impl<Input> InputResolver<Input> {
+    pub fn new(plan: ExecutionPlan<Input>) -> Self {
+        Self { plan }
+    }
+}
+
+impl<Input, Output> InputResolver<Input>
```

Comment:
I don't think is really needed. We can implement this behaviour by adding a `try_map` in `ExecutionPlan`.

Reply:
when we discussed this you suggested to implement value resolution as separate step e.g. to have Field<Input> -> Field<ResolvedInput> conversion